src/serverless/structure/serverless_parser.go

```diff
@@ -0,0 +1,317 @@
+package structure
+
+import (
+	"bridgecrewio/yor/src/common"
+	"bridgecrewio/yor/src/common/logger"
+	"bridgecrewio/yor/src/common/structure"
+	"bridgecrewio/yor/src/common/tagging/tags"
+	"bufio"
+	"fmt"
+	"github.com/awslabs/goformation/v4/cloudformation"
+	"io"
+	"math"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+)
+
+const FunctionTagsAttributeName = "tags"
+const StackTagsAttributeName = "stackTags"
+
+var TemplateSections = []string{"service", "provider", "tags", "stackTags", "resources", "functions", "region", "Resources"}
+
+type ServerlessTemplate struct {
+	Service  string `yaml:"service"`
+	Provider struct {
+		Name         string            `yaml:"name"`
+		Runtime      string            `yaml:"runtime"`
+		Region       string            `yaml:"region"`
+		ProviderTags map[string]string `yaml:"tags"`
+		CFNTags      map[string]string `yaml:"stackTags"`
+		Functions    interface{}       `yaml:"functions"`
+		Resources    struct {
+			Resources *cloudformation.Template `yaml:"Resources"`
+		} `yaml:"resources"`
+	} `yaml:"provider"`
+}
+
+type ServerlessParser struct {
+	rootDir              string
+	fileToResourcesLines map[string]common.Lines
+	template             *ServerlessTemplate
+}
+
+func (p *ServerlessParser) Init(rootDir string, _ map[string]string) {
+	p.rootDir = rootDir
+	p.fileToResourcesLines = make(map[string]common.Lines)
+	p.template = &ServerlessTemplate{}
+}
+
+func (p *ServerlessParser) GetSkippedDirs() []string {
+	return []string{}
+}
+
+func (p *ServerlessParser) GetSupportedFileExtensions() []string {
+	return []string{common.YamlFileType.Extension, common.YmlFileType.Extension}
+}
+
+func (p *ServerlessParser) ParseFile(filePath string) ([]structure.IBlock, error) {
+	parsedBlocks := make([]structure.IBlock, 0)
+	template, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		logger.Warning(fmt.Sprintf("There was an error processing the serverless template: %s", err))
+	}
+	err = yaml.Unmarshal([]byte(template), p.template)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Unmarshal: %s", err), "SILENT")
+	}
+
+	if err != nil || template == nil {
+		logger.Error(fmt.Sprintf("There was an error processing the serverless template: %s", err), "SILENT")
+
+	}
+	//cfnStackTagsResource := p.template.Provider.CFNTags
+	functions := p.template.Provider.Functions
+	functionsMap := functions.(map[interface{}]interface{})
+	resourceNames := make([]string, 0)
```

Comment:
@tronxd Isn't `functionsNames` is a better name for this variable?

Reply:
Not in my opinion. they are tagged resources